site-src/geps/gep-995.md

```diff
@@ -0,0 +1,66 @@
+# GEP-955: HTTP Route Naming
+
+* Issue: [#995](https://github.com/kubernetes-sigs/gateway-api/issues/995)
+* Status: Implementable
+
+## TLDR
+
+* Add `Name` field to `Rules` field in HTTPRoute.
+* Add `Name` field to `Matches` field in HTTPRouteRule.
+
+## Goals
+
+Provide a path for implementations to support:
+
+* Policy Attachment:
+  Attach policies to HTTP route.
+* Route Attachment/Delegation:
+  Ability to have more fine-grained matches for policy references
+  if we add sectionName to the policy object.
+* Improved Debugging: 
+  Ability to add debugging information, such as adding the route matched to some access log.
+
+## Non-Goals
+
+* Define how implementations should support these features.
+
+## Introduction
+
+As described above, there are a number of potential use cases for providing an identity to HTTP route.
+The most straightforward reason for this is to allow extending a particular HTTP route.
+This change will enable implementations to support the above use cases while still providing a portable core.
+
+## API
+
+A `Name` field would be added to `HTTPRouteMatch`:
+
+```go
+type HTTPRouteRule struct {
+    // The name assigned to the route for debugging purposes. The
+    // route's name will be concatenated with the match's name and may
+    // be logged in the access logs for requests matching this
+    // route/match depends on the implementor.
+    //
+    // +optional
+    Name *string `json:"name,omitempty"`
+    ...
+}
+```
+
+
+A `Name` field would be added to `HTTPRouteRule `:
+
+```go
+type HTTPRouteMatch struct {
+    // Name specifies the HTTP route match name. The match's name will be
+    // concatenated with the parent route's name and may be logged in
+    // the access logs for requests matching this route depends on the implementor.
+    //
+    // +optional
+    Name *string `json:"name,omitempty"`
+	...
+}
+```
+## Alternatives
+
+N/A
```

Comment:
I think there's a few alternatives here:

1. Users can split out Routes into different resources when they need to be referenced separately.
Route rules are really more of a convenience than a necessity. Instead of having 10 rules in 1 route, the same config could be represented by 10 routes with 1 rule each.

2. Route attachment/delegation may choose to use a different approach.
For example, each Route could choose to include a set of child routes by direct reference, meaning this GEP would not be useful for that approach.

3. Logging could assign numerical indexes to Route rules.
Instead of `rules[sectionName]` logs would include `rules[0]` or similar.